core/src/main/java/org/apache/metamodel/QueryPostprocessDataContext.java

```diff
@@ -406,18 +409,15 @@ protected DataSet materializeTable(final Table table, final List<SelectItem> sel
     }
 
     private List<SelectItem> buildWorkingSelectItems(List<SelectItem> selectItems, List<FilterItem> whereItems) {
-        final List<SelectItem> primarySelectItems = new ArrayList<>(selectItems.size());
-        for (SelectItem selectItem : selectItems) {
```

Comment:
Just to say it ... This is one of the things that I am really pleased to get rid of. Because the `MetaModelHelper.getFiltered` and `MetaModelHelper.getSelection` methods are now taking care of scalar functions, stuff like this for-loop is no longer needed.